pyard/ard.py

```diff
@@ -253,6 +254,20 @@ def _redux_allele(
             else:
                 # If ambiguous, reduce to G group level
                 return self._redux_allele(allele, "lgx")
+        elif redux_type == "S":
+            # find serology equivalent in serology_mapping
+            serology_mapping = db.find_serology_for_allele(self.db_connection, allele)
+            serology_set = set()
+            if is_2_field_allele(allele):
+                for serology, allele_list in serology_mapping.items():
+                    allele_list_lgx = self.redux(allele_list, "lgx")
+                    if allele in allele_list_lgx:
+                        serology_set.add(serology)
+            else:
+                for serology, allele_list in serology_mapping.items():
+                    if allele in allele_list:
+                        serology_set.add(serology)
+            return "/".join(serology_set)
```

Comment:
I think we should sort the result before the join

Reply:
👍 Serology set is now sorted.